javascript/ql/lib/semmle/javascript/internal/ConceptsShared.qll

```diff
@@ -43,6 +43,9 @@ module Cryptography {
     /** Gets an input the algorithm is used on, for example the plain text input to be encrypted. */
     DataFlow::Node getAnInput() { result = super.getAnInput() }
 
+    /** DEPRECATED. This predicate has been renamed to `getAnInput`. */
+    deprecated final DataFlow::Node getInput() { result = super.getInput() }
```

Comment:
This will cause compile-errors for any external users that have extended this.  
(Also see my former comment about `CryptographicOperation` and change-note).  

I think we'll have to use `breaking` in the change-note. 

I think the QLDoc should loudly explain what to do, which is both extend `getAnInput` instead, **and** make sure you extend `CryptographicOperation::Range` instead of `CryptographicOperation`.

Reply:
I've updated the QLDoc to mention this (the predicate itself has moved to `Concepts.qll` to avoid introducing a deprecated predicate to Ruby/Py as well). It's a little bit clumsy as the deprecation message has to read properly for both users and implementers of `CryptographicOperation#getInput()`.